### PR TITLE
Previews are fun for everybody.

### PR DESCRIPTION
**Pull request by BjornTwachtmann**
https://jira.dev.bbc.co.uk/browse/BLOGS-1405

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/BlogsService/Service/PostService.php [hunk 1]
```diff
@@ -38,18 +38,19 @@ public function __construct(
     public function getPostByGuid(
         GUID $guid,
         ?Blog $blog = null,
+        bool $preview,
```
Comment (Tasiobg) [inline src/BlogsService/Service/PostService.php]: Code linting tools think is a good practice to put optional parameters after required ones, I'm not too sure if is just a fancy thing or if there is real value on it

File: src/Controller/BaseController.php [hunk 2]
```diff
@@ -136,6 +138,17 @@ protected function setBrandingId(string $brandingId)
         }
     }
 
+    protected function isPreview(Request $request): bool
+    {
+        $preview = filter_var($request->get('preview', 'false'), FILTER_VALIDATE_BOOLEAN);
+        return (bool) $preview;
+    }
+
+    protected function setMetaNoIndex(bool $metaNoIndex)
```
Comment (Tasiobg) [inline src/Controller/BaseController.php]:
```suggestion
    protected function setMetaNoIndex(bool $metaNoIndex): void
```

File: src/Controller/CommentsPartialController.php [hunk 3]
```diff
@@ -10,17 +10,18 @@
 use BBC\ProgrammesMorphLibrary\Exception\MorphErrorException;
 use Exception;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 
-class CommentsPartialController extends AbstractController
+class CommentsPartialController extends BaseController
```
Comment (Tasiobg) [inline src/Controller/CommentsPartialController.php]: you can remove `use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;`

File: src/ValueObject/MetaContext.php [hunk 4]
```diff
@@ -12,6 +12,19 @@
 
 class MetaContext
 {
+    /** @var bool */
```
Comment (Tasiobg) [inline src/ValueObject/MetaContext.php]: Not part of this PR but probably worth fixing it here, I don't think any of the `use` of this file makes any sense, feel free to remove them
```
use BBC\ProgrammesPagesService\Domain\Entity\CoreEntity;
use BBC\ProgrammesPagesService\Domain\Entity\Image;
use BBC\ProgrammesPagesService\Domain\Entity\Network;
use BBC\ProgrammesPagesService\Domain\Entity\Programme;
use BBC\ProgrammesPagesService\Domain\Entity\Service;
use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
```

**Comment by Tasiobg** [comment]
I haven't tested it. Does it requires something like this in blogs as well?
https://github.com/bbc/programmes-frontend/blob/develop/src/Controller/ExceptionController.php#L85

**Comment by BjornTwachtmann** [comment]
> I haven't tested it. Does it requires something like this in blogs as well?
> https://github.com/bbc/programmes-frontend/blob/develop/src/Controller/ExceptionController.php#L85

Shit, that's a good point, completely forgot about that

**Pull request merged**